odoo/addons/base/models/res_partner.py

```diff
@@ -147,6 +148,11 @@ def _name_search(self, name, domain=None, operator='ilike', limit=None, order=No
             domain = [('name', operator, name)] + domain
         return self._search(domain, limit=limit, order=order)
 
+    @api.model
+    def _search_display_name(self, operator, value, limit=None):
```

Comment:
Would try a filter_domain in view instead of hacking the display_name search (as it is not respecting operator, quite not standard, ...).